ASAN patch breaks kernel built-in builds

reverting fed90353d799acbc5e81b0dfadc6d649b0f2e8b5 'fixes' it.

```
  CC      fs/zfs/zcommon/zfeature_common.o
fs/zfs/zcommon/zfeature_common.c:32:19: fatal error: errno.h: No such file or directory
 #include <errno.h>
                   ^
compilation terminated.
```

```
In file included from ./include/zfs/sys/edonr.h:43:0,
                 from fs/zfs/icp/algs/edonr/edonr.c:38:
/usr/lib/gcc/x86_64-pc-linux-gnu/6.4.0/include/stdint.h:9:26: error: no include path in which to search for stdint.h
 # include_next <stdint.h>
                          ^
In file included from fs/zfs/icp/algs/edonr/edonr.c:38:0:
./include/zfs/sys/edonr.h:44:37: fatal error: stdlib.h: No such file or directory
 #include <stdlib.h> /* size_t ... */
                                     ^
compilation terminated.
make[3]: *** [scripts/Makefile.build:315: fs/zfs/icp/algs/edonr/edonr.o] Error 1
make[3]: *** Waiting for unfinished jobs....
make[2]: *** [scripts/Makefile.build:573: fs/zfs/icp] Error 2
make[1]: *** [scripts/Makefile.build:573: fs/zfs] Error 2
make: *** [Makefile:1028: fs] Error 2
make: *** Waiting for unfinished jobs....
```

It seems like include paths are absent?

that certianly helps though now I'm getting the following


```
# make -j4 && make modules_install
  CHK     include/config/kernel.release
  CHK     include/generated/uapi/linux/version.h
  DESCEND  objtool
  CHK     include/generated/utsrelease.h
  CHK     scripts/mod/devicetable-offsets.h
  CHK     include/generated/timeconst.h
  CHK     include/generated/bounds.h
  CHK     include/generated/asm-offsets.h
  CALL    scripts/checksyscalls.sh
  CHK     include/generated/compile.h
  CHK     kernel/config_data.h
  CC      fs/zfs/zcommon/zfeature_common.o
fs/zfs/zcommon/zfeature_common.c: In function ‘zfeature_is_supported’:
fs/zfs/zcommon/zfeature_common.c:97:2: error: ‘for’ loop initial declarations are only allowed in C99 or C11 mode
  for (spa_feature_t i = 0; i < SPA_FEATURES; i++) {
  ^~~
fs/zfs/zcommon/zfeature_common.c:97:2: note: use option -std=c99, -std=gnu99, -std=c11 or -std=gnu11 to compile your code
fs/zfs/zcommon/zfeature_common.c: In function ‘zfeature_lookup_name’:
fs/zfs/zcommon/zfeature_common.c:108:2: error: ‘for’ loop initial declarations are only allowed in C99 or C11 mode
  for (spa_feature_t i = 0; i < SPA_FEATURES; i++) {
  ^~~
fs/zfs/zcommon/zfeature_common.c: In function ‘zfeature_depends_on’:
fs/zfs/zcommon/zfeature_common.c:125:2: error: ‘for’ loop initial declarations are only allowed in C99 or C11 mode
  for (int i = 0; feature->fi_depends[i] != SPA_FEATURE_NONE; i++) {
  ^~~
fs/zfs/zcommon/zfeature_common.c: In function ‘deps_contains_feature’:
fs/zfs/zcommon/zfeature_common.c:135:2: error: ‘for’ loop initial declarations are only allowed in C99 or C11 mode
  for (int i = 0; deps[i] != SPA_FEATURE_NONE; i++)
  ^~~
make[3]: *** [scripts/Makefile.build:314: fs/zfs/zcommon/zfeature_common.o] Error 1
make[2]: *** [scripts/Makefile.build:573: fs/zfs/zcommon] Error 2
make[1]: *** [scripts/Makefile.build:573: fs/zfs] Error 2
make: *** [Makefile:1028: fs] Error 2
make: *** Waiting for unfinished jobs....
```

better I think, but still some fail.  I had to remove the following line to get it to work at all as well.

    +       ZFS_MODULE_CPPFLAGS += @KERNEL_DEBUG_CPPFLAGS@

```
fs/zfs/zfs/dbuf.c: In function ‘dbuf_evict_user’:
fs/zfs/zfs/dbuf.c:405:2: warning: ISO C90 forbids mixed declarations and code [-Wdeclaration-after-statement]
  boolean_t has_async = (dbu->dbu_evict_func_async != NULL);
  ^~~~~~~~~
fs/zfs/zfs/dbuf.c: In function ‘dbuf_evict_one’:
fs/zfs/zfs/dbuf.c:515:2: warning: ISO C90 forbids mixed declarations and code [-Wdeclaration-after-statement]
  dmu_buf_impl_t *db = multilist_sublist_tail(mls);
  ^~~~~~~~~~~~~~
fs/zfs/zfs/dbuf.c: In function ‘dmu_buf_will_dirty_impl’:
fs/zfs/zfs/dbuf.c:1990:2: warning: ISO C90 forbids mixed declarations and code [-Wdeclaration-after-statement]
  dbuf_dirty_record_t *dr;
  ^~~~~~~~~~~~~~~~~~~
fs/zfs/zfs/dbuf.c: In function ‘dbuf_findbp’:
fs/zfs/zfs/dbuf.c:2329:2: warning: ISO C90 forbids mixed declarations and code [-Wdeclaration-after-statement]
  int nlevels =
  ^~~
fs/zfs/zfs/dbuf.c: In function ‘dbuf_issue_final_prefetch’:
fs/zfs/zfs/dbuf.c:2502:2: warning: ISO C90 forbids mixed declarations and code [-Wdeclaration-after-statement]
  arc_flags_t aflags =
  ^~~~~~~~~~~
fs/zfs/zfs/dbuf.c: In function ‘dbuf_prefetch_indirect_done’:
fs/zfs/zfs/dbuf.c:2565:2: warning: ISO C90 forbids mixed declarations and code [-Wdeclaration-after-statement]
  uint64_t nextblkid = dpa->dpa_zb.zb_blkid >>
  ^~~~~~~~
fs/zfs/zfs/dbuf.c: In function ‘dbuf_prefetch’:
fs/zfs/zfs/dbuf.c:2634:2: warning: ISO C90 forbids mixed declarations and code [-Wdeclaration-after-statement]
  dmu_buf_impl_t *db = dbuf_find(dn->dn_objset, dn->dn_object,
  ^~~~~~~~~~~~~~
fs/zfs/zfs/dbuf.c:2679:2: warning: ISO C90 forbids mixed declarations and code [-Wdeclaration-after-statement]
  zio_t *pio = zio_root(dmu_objset_spa(dn->dn_objset), NULL, NULL,
  ^~~~~
In file included from ./include/linux/kernel.h:10:0,
                 from ./include/linux/list.h:9,
                 from ./include/linux/module.h:9,
                 from ./include/spl/sys/sysmacros.h:28,
                 from ./include/spl/sys/types.h:29,
                 from ./include/zfs/sys/zfs_context.h:34,
                 from fs/zfs/zfs/dbuf.c:29:
fs/zfs/zfs/dbuf.c: In function ‘__dbuf_hold_impl’:
fs/zfs/zfs/dbuf.c:2829:38: error: implicit declaration of function ‘arc_referenced’ [-Werror=implicit-function-declaration]
  ASSERT(dh->dh_db->db_buf == NULL || arc_referenced(dh->dh_db->db_buf));
                                      ^
./include/linux/compiler.h:77:42: note: in definition of macro ‘unlikely’
 # define unlikely(x) __builtin_expect(!!(x), 0)
                                          ^
./include/spl/sys/debug.h:108:23: note: in expansion of macro ‘VERIFY’
 #define ASSERT(cond)  VERIFY(cond)
                       ^~~~~~
fs/zfs/zfs/dbuf.c:2829:2: note: in expansion of macro ‘ASSERT’
  ASSERT(dh->dh_db->db_buf == NULL || arc_referenced(dh->dh_db->db_buf));
  ^~~~~~
fs/zfs/zfs/dbuf.c: In function ‘dbuf_write’:
fs/zfs/zfs/dbuf.c:3988:3: warning: ISO C90 forbids mixed declarations and code [-Wdeclaration-after-statement]
   arc_write_done_func_t *children_ready_cb = NULL;
   ^~~~~~~~~~~~~~~~~~~~~
```

fs/zfs/zfs/dbuf.c:2829:38: error: implicit declaration of function ‘arc_referenced’ [-Werror=implicit-function-declaration]
      ASSERT(dh->dh_db->db_buf == NULL || arc_referenced(dh->dh_db->db_buf));

It looks like that function is only there if ZFS_DEBUG is set.

```
#ifdef ZFS_DEBUG
int
arc_referenced(arc_buf_t *buf)
{
        int referenced;

        mutex_enter(&buf->b_evict_lock);
        referenced = (refcount_count(&buf->b_hdr->b_l1hdr.b_refcnt));
        mutex_exit(&buf->b_evict_lock);
        return (referenced);
}
#endif
```

after that's un-debug'd

```
In file included from ./include/linux/kernel.h:10:0,
                 from ./include/linux/list.h:9,
                 from ./include/linux/module.h:9,
                 from ./include/spl/sys/sysmacros.h:28,
                 from ./include/spl/sys/types.h:29,
                 from ./include/zfs/sys/zfs_context.h:34,
                 from fs/zfs/zfs/dnode.c:27:
fs/zfs/zfs/dnode.c: In function ‘dnode_rele_and_unlock’:
fs/zfs/zfs/dnode.c:1501:36: error: ‘zrlock_t {aka struct zrlock}’ has no member named ‘zr_owner’
  ASSERT(refs > 0 || dnh->dnh_zrlock.zr_owner != curthread);
                                    ^
./include/linux/compiler.h:77:42: note: in definition of macro ‘unlikely’
 # define unlikely(x) __builtin_expect(!!(x), 0)
                                          ^
./include/spl/sys/debug.h:108:23: note: in expansion of macro ‘VERIFY’
 #define ASSERT(cond)  VERIFY(cond)
                       ^~~~~~
fs/zfs/zfs/dnode.c:1501:2: note: in expansion of macro ‘ASSERT’
  ASSERT(refs > 0 || dnh->dnh_zrlock.zr_owner != curthread);
  ^~~~~~
```

--- a/copy-builtin
+++ b/copy-builtin
@@ -63,8 +63,10 @@ EOF
 	ZFS_MODULE_CFLAGS += -I$(srctree)/include/spl
 	ZFS_MODULE_CFLAGS += -include $(srctree)/spl_config.h
 	ZFS_MODULE_CFLAGS += -include $(srctree)/zfs_config.h
-	ZFS_MODULE_CLFAGS += -std=gnu99
-	export ZFS_MODULE_CFLAGS
+	ZFS_MODULE_CFLAGS += -std=gnu99 -Wno-declaration-after-statement
+	ZFS_MODULE_CPPFLAGS  = -DHAVE_SPL -D_KERNEL
+	ZFS_MODULE_CPPFLAGS += -UDEBUG -DNDEBUG
+	export ZFS_MODULE_CFLAGS ZFS_MODULE_CPPFLAGS
 
 	obj-$(CONFIG_ZFS) :=
 	EOF

